nemo/collections/speechlm2/parts/hf_hub.py

```diff
@@ -69,6 +69,11 @@ def _from_pretrained(
         # this setting skips loading the original pretrained ASR and LLM weights, and loads the
         # final trained model weights directly.
         model_kwargs['cfg']['pretrained_weights'] = False
+        # When loading from HF checkpoint, set init_from_path to the HF checkpoint directory
+        # so it can load the model weights from the correct location.
+        if 'init_from_path' in model_kwargs['cfg']:
```

Comment:
`init_from_path` is used to load a checkpoint from a previous stage. 
In Stage 1, this argument is not set.
In Stage 2, it points to the final HF checkpoint from Stage 1.
When loading the final checkpoint after Stage 2, perhaps we do not need `init_from_path`, assuming the HF checkpoint saves everything.